### ISSUE TITLE
[v2] drop support for legacy CommonJS

### ISSUE THREAD

**Issue by trusktr** [PrismJS/prism#3703]
We only need to support ES Modules these days. All major JS engines (Node, Deno, and all browsers) support native ES Modules.

To make maintenance simpler, we can simply output the source as ES modules and leave it at that.

Anyone who _really_ needs CommonJS (f.e. they are stuck with some _really old_ build tool for some reason) can get Rollup and compile a CommonJS bundle that they can use in their case. These days, those people are in a minority, and quickly dwindling away.

**Comment by trusktr** [comment]
I just noticed the `package.json` already states:

```json
	"engines": {
		"node": ">=14"
	},
```

so we already require Node with ESM. People with legacy CommonJS code can `await import('...')`.

**Comment by maximebochon** [comment]
I agree, I get this warning when depending on `prism-copy-to-clipboard'`:

```
Warning: [...] depends on 'prismjs/plugins/copy-to-clipboard/prism-copy-to-clipboard'.
CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
```
